samples/bot-tag-mention/csharp/Dialogs/MainDialog.cs

```diff
@@ -0,0 +1,111 @@
+{
+    public class MainDialog : LogoutDialog
+    {
+        protected readonly ILogger _logger;
+
+        public MainDialog(IConfiguration configuration, ILogger<MainDialog> logger)
+            : base(nameof(MainDialog), configuration["ConnectionName"])
+        {
+            _logger = logger;
+
+            AddDialog(new OAuthPrompt(
+                nameof(OAuthPrompt),
+                new OAuthPromptSettings
+                {
+                    ConnectionName = ConnectionName,
+                    Text = "Please Sign In",
+                    Title = "Sign In",
+                    Timeout = 300000, // User has 5 minutes to login (1000 * 60 * 5)
+                    EndOnInvalidMessage = true
+                }));
+
+            AddDialog(new ConfirmPrompt(nameof(ConfirmPrompt)));
+
+            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
+            {
+                PromptStepAsync,
+                MentionTagAsync
+            }));
+
+            // The initial child Dialog to run.
+            InitialDialogId = nameof(WaterfallDialog);
+        }
+
+        private async Task<DialogTurnResult> PromptStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("PromptStepAsync() called.");
+            return await stepContext.BeginDialogAsync(nameof(OAuthPrompt), null, cancellationToken);
+        }
+
+        private async Task<DialogTurnResult> MentionTagAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            // Get the token from the previous step. Note that we could also have gotten the
+            // token directly from the prompt itself. There is an example of this in the next method.
+            var tokenResponse = (TokenResponse)stepContext.Result;
+            if (tokenResponse?.Token != null)
+            {
+                try
+                {
+                    stepContext.Context.Activity.RemoveRecipientMention();
+                    if (stepContext.Context.Activity.Text.Trim().ToLower() == "mentiontag")
+                    {
+                        // Pull in the data from the Microsoft Graph.
+                        var client = new SimpleGraphClient(tokenResponse.Token);
+                        var teamID = await TeamsInfo.GetTeamDetailsAsync(stepContext.Context, stepContext.Context.Activity.TeamsGetTeamInfo().Id, cancellationToken);
+                        var result = await client.GetTag(teamID.AadGroupId);
+                        var adaptiveCardTemplate = Path.Combine(".", "Resources", "UserMentionCardTemplate.json");
+                        var templateJSON = File.ReadAllText(adaptiveCardTemplate);
+                        AdaptiveCardTemplate template = new AdaptiveCardTemplate(templateJSON);
+                        var memberData = new
+                        {
+                            tagId = result.CurrentPage[0].Id,
```

Comment:
can you add the logic to check whether there is existing tag in current team before using CurrentPage[0]?

Reply:
Added changes to the code with all necessary exception handling features.